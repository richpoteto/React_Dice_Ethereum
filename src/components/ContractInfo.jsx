import React from 'react';
import PropTypes from 'prop-types';
import Address from './Address';

function ContractInfo({ accountAddress, contractAddress, network }) {
  const contractAddr = <Address network={network} address={contractAddress} />;
  const contract = (
    <div className="col-lg-6">
      Contract
      &nbsp;
      {contractAddr}
    </div>
  );
  const accountAddr = (accountAddress !== null)
    ? <Address network={network} address={accountAddress} />
    : <span>Not connected, please login to MetaMask</span>;
  const account = (
    <div className="col-lg-6">
      Account
      &nbsp;
      {accountAddr}
    </div>
  );
  return (
    <div className="row">
      {contract}
      {account}
    </div>
  );
}
ContractInfo.propTypes = {
  accountAddress: PropTypes.string,
  contractAddress: PropTypes.string.isRequired,
  network: PropTypes.number.isRequired,
};
ContractInfo.defaultProps = {
  accountAddress: null,
};

export default ContractInfo;
