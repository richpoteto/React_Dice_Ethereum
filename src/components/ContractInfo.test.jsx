import React from 'react';
import renderer from 'react-test-renderer';
import ContractInfo from './ContractInfo';

it('renders correctly', () => {
  const tree = renderer
    .create(<ContractInfo accountAddress="0x0123" contractAddress="0x3210" network={3} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});


it('no account address', () => {
  const tree = renderer
    .create(<ContractInfo contractAddress="0x3210" network={3} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
