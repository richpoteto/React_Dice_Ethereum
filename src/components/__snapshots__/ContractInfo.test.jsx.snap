// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`no account address 1`] = `
<div
  className="row"
>
  <div
    className="col-lg-6"
  >
    Contract  
    <a
      href="https://ropsten.etherscan.io/address/0x3210"
    >
      0x3210
    </a>
  </div>
  <div
    className="col-lg-6"
  >
    Account  
    <span>
      Not connected, please login to MetaMask
    </span>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="row"
>
  <div
    className="col-lg-6"
  >
    Contract  
    <a
      href="https://ropsten.etherscan.io/address/0x3210"
    >
      0x3210
    </a>
  </div>
  <div
    className="col-lg-6"
  >
    Account  
    <a
      href="https://ropsten.etherscan.io/address/0x0123"
    >
      0x0123
    </a>
  </div>
</div>
`;
